pkgs/build-support/setup-hooks/auto-patchelf.sh

```diff
@@ -187,14 +248,21 @@ autoPatchelfFile() {
     # new package where you don't yet know its dependencies.
 
     for dep in $missing; do
-        # Check whether this library exists in libc. If so, we don't need to do
-        # any futher searching -- it will be resolved correctly by the linker.
-        if [ -f "$libcLib/$dep" ]; then
+        if [[ "$dep" == /* ]]; then
+            # This is an absolute path. If it exists, just use it. Otherwise,
+            # we probably want this to produce an error when checked (because
+            # just updating the rpath won't satisfy it).
+            if [ -f "$dep" ]; then
+                continue
+            fi
+        elif [ -f "$libcLib/$dep" ]; then
```

Comment:
Why are we mixing [ and [[?

Reply:
I prefer to always use `[[` myself when I'm guaranteed to have a bash shell, but I was trying to match the existing style of the file which seemed to only use `[[` when it was required and `[` otherwise. Maybe I was reading too much into it though. :-)

Since this was already merged, would you like me to open another PR for some of these stylistic fixes?

Reply:
NixOS's stdenv is guaranteed to use bash. I wouldn't worry personally about portability at all. You can keep it like it is for now but in the future I think it would be better to either fully use [ or [[ and not mix and match them.

Reply:
That makes sense, thanks for the clarification.

I do want to introduce one more change to this script, which is to bring back #66620 so that packages that need `stdenv_32bit` will work on 64-bit. So when I get around to that I'll include these suggestions as well. Thanks again!